kafka/common.go

```diff
@@ -215,6 +216,21 @@ func (cfg *CommonConfig) finalize() error {
 		if os.Getenv("KAFKA_TLS_INSECURE") == "true" {
 			cfg.TLS.InsecureSkipVerify = true
 		}
+		if caCertPath := os.Getenv("KAFKA_TLS_CA_CERT_PATH"); caCertPath != "" {
+			caCert, err := os.ReadFile(caCertPath)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("kafka: failed to read CA cert: %w", err))
+			} else {
+				rootCa, err := x509.SystemCertPool()
```

Comment:
I wonder if we should _only_ accept the configured CA? Then clients would have more control

Reply:
You would not be at the mercy of root CAs, which can be compromised. Rare I know, but it happens, e.g. https://blog.mozilla.org/security/2011/03/25/comodo-certificate-issue-follow-up/